apps/src/code-studio/components/libraries/LibraryPublisher.jsx

```diff
@@ -0,0 +1,224 @@
+      error => {
+        console.warn(`Error publishing library: ${error}`);
+        this.setState({publishState: PublishState.ERROR_PUBLISH});
+      },
+      () => {
+        onPublishSuccess(libraryName);
+      }
+    );
+    dashboard.project.setLibraryName(libraryName);
+    dashboard.project.setLibraryDescription(libraryDescription);
+  };
+
+  displayNameInput = () => {
+    return (
+      <div>
+        <input
+          style={styles.textInput}
+          type="text"
+          value={this.state.libraryName}
+          onChange={this.setLibraryName}
+          onBlur={event =>
+            this.setState({
+              libraryName: libraryParser.suggestName(event.target.value)
+            })
+          }
+        />
+        <div style={styles.info}>{i18n.libraryNameRequirements()}</div>
+      </div>
+    );
+  };
+
+  resetErrorMessage = () => {
+    if (
+      this.state.libraryDescription &&
+      Object.values(this.state.selectedFunctions).find(value => value) &&
+      this.state.publishState === PublishState.INVALID_INPUT
+    ) {
+      this.setState({publishState: PublishState.DEFAULT});
+    }
+  };
+
+  displayDescription = () => {
+    return (
+      <textarea
+        rows="2"
+        cols="200"
+        style={{...styles.textInput, ...styles.description}}
+        placeholder={i18n.libraryDescriptionPlaceholder()}
+        value={this.state.libraryDescription}
+        onChange={event => {
+          this.setState(
+            {libraryDescription: event.target.value},
+            this.resetErrorMessage
+          );
+        }}
+      />
+    );
+  };
+
+  boxChecked = name => () => {
+    this.setState(state => {
+      state.selectedFunctions[name] = !state.selectedFunctions[name];
+      return state;
+    }, this.resetErrorMessage);
+  };
+
+  displayFunctions = () => {
+    return this.props.libraryDetails.sourceFunctionList.map(sourceFunction => {
+      let name = sourceFunction.functionName;
+      let comment = sourceFunction.comment;
+      return (
+        <div key={name}>
+          <input
+            style={styles.largerCheckbox}
+            type="checkbox"
+            disabled={comment.length === 0}
+            name={name}
+            checked={this.state.selectedFunctions[name] || false}
+            onChange={this.boxChecked(name)}
+          />
+          <span>{name}</span>
+          <br />
+          {comment.length === 0 && (
+            <p style={styles.alert}>{i18n.libraryExportNoCommentError()}</p>
+          )}
+          <pre style={styles.textInput}>{comment}</pre>
+        </div>
+      );
+    });
+  };
+
+  displayError = () => {
+    let errorMessage;
+    if (this.state.publishState === PublishState.INVALID_INPUT) {
+      errorMessage = i18n.libraryPublishInvalid();
+    }
+    if (this.state.publishState === PublishState.ERROR_PUBLISH) {
+      errorMessage = i18n.libraryPublishFail();
+    }
+    return (
+      <div>
+        <p style={styles.alert}>{errorMessage}</p>
```

Comment:
It looks like this will be rendered even when there is not error message?

Reply:
✔️